thanks-bot/bot.js

```diff
@@ -11,42 +12,46 @@ logger.add(logger.transports.Console, {
 logger.level = 'debug'
 
 // Initialize Discord Bot
-const bot = new Discord.Client({
-  token: auth.token,
-  autorun: true
-})
+const client = new Discord.Client()
 
-bot.on('ready', function (evt) {
+client.on('ready', () => {
   logger.info('Connected')
   logger.info('Logged in as: ')
-  logger.info(bot.username + ' - (' + bot.id + ')')
+  logger.info(`${client.user.username} - (${client.user.id})`)
 })
 
 const ALLOWED_CHANNEL_ID = '525249065039036426'
 
-bot.on('message', function (user, userID, channelID, message, evt) {
-  let thanksMessageRegexp = /!thanks\b/gmi
+client.on('message', message => {
+  const thanksMessageRegexp = /!thanks\b/gmi
 
   // Conditions for exiting the function
-  if (channelID !== ALLOWED_CHANNEL_ID ||
-    userID === bot.id || !thanksMessageRegexp.test(message)) {
+  if (message.channel.id !== ALLOWED_CHANNEL_ID ||
+    message.author.id === client.user.id ||
+    message.content !== thanksMessageRegexp.test(message)) {
```

Comment:
In the previous code it was just `!thanksMessageRegexp.test(message)`, now it's `message.content !== thanksMessageRegexp.test(message)`.
The thing is that [`test` returns either `true` or `false`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test). When you then compare it with strings, in all cases you get `false`.